### PR TITLE
 Fix QA Pulls Tool merge instructions

### PR DESCRIPTION
**Pull request by carusogabriel**
As suggested by @cmb69 in https://github.com/php/php-src/pull/3613#issuecomment-430927108.

/cc @petk @johannes

### CODE REVIEW

File: pulls/index.php [hunk 1]
```diff
@@ -84,12 +84,13 @@
    </script>
    <script id="pullInstructionTemplate" type="text/x-jquery-tmpl">
 	   <pre>
-$ git fetch git://github.com/php/{{=repo}} pull/{{=number}}/head:pull-request/{{=number}}
-$ git log -p pull-request/{{=number}} # REVIEW IT
-$ git merge pull-request/{{=number}}  # Merge it, add a GOOD commit message
-$ make test                  # you better not forget that
-$ git push origin master     # everything okay? good, let's push it
+$ git checkout master
```
Comment (carusogabriel) [inline pulls/index.php]: @johannes Is there a way to know the branch the author of the PR wants to merge into? We could use it here to improve this command :)

**Comment by petk** [comment]
Hello, thanks for taking a look at this one. I was just thinking about this... GitHub seems to have a way to do a merge without a merge commit in their buttons - maybe something to investigate a bit more how this works exactly. I understand that they have more info which pull request and particular commits are connected to each other and command line might not. Is this something possible in the command line as well by any other chance?

If not, then the raw way is ok I guess... These merge instructions are still cool to know for some other projects that might utilize them... I'll copy/paste them in my cheat sheet :D

File: pulls/index.php (hunk 1 above)
Comment (petk) [reply pulls/index.php]: It's in the info of https://api.github.com/repos/php/php-src/pulls as base.ref ... and then the `pulls/pullrequest.js` needs to be adjusted a bit at line 160 by passing a new template element.

**Comment by php-pulls** [comment]
**Comment on behalf of petk at php.net:**

Applied via c3e1097e314c7490c9a466f4226cfab844854902

**Comment by johannes** [comment]
I'm not sure this approach really is the nice way to do it. Makes it hard to manage the pull request if the author updates it after feedback. Also I would always suggest to work in a names branch instead of master, as test/feedback/fix cycles might take some time. Wouldm't it be enough to suggest a rebase in the end? (And remind about other branches, cleanup etc.)

**Comment by johannes** [comment]
And for integration with newer GitHub features we have to see ... from my perspective (not that I'm very active in PHP these days, so no idea how much it counts) I think our process should not be too dependent on an external entity like GitHub. It is good to use them, but we have seen platforms and even version control systems come and go over PHP's lifetime.

**Comment by johannes** [comment]
So a `git checkout pull/1234 && git rebase master` might actually be the way, which stays in the tool?

**Comment by cmb69** [comment]
> So a git checkout pull/1234 && git rebase master might actually be the way, which stays in the tool?

Afterwards a `git merge` would still be necessary, wouldn't it? See https://github.com/php/php-src/pull/2835#issuecomment-335939245 ff. regarding why applying the patch is preferred.

**Comment by petk** [comment]
Ideal procedure would in steps be something like the following:
- fetch remote branch from the pull request (so no patches are needed to be downloaded)
- squash all commits of the pull request branch into a single contribution (if there was a single contributor working on this), otherwise apply multiple commits (GitHub has now additional features that would be lovely to get utilized - like suggestions and these can have multiple commits from different authors)
- git merge the pull request branch without additional merge commit
- git push

And in the end have no merge commit and pull request merged in a violet color. Not closed, in red...

**Pull request closed without merging**